pkg/controller/dns/external_dns_crd.go

```diff
@@ -0,0 +1,118 @@
+		})
+}
+
+func (e *ExternalDNSCRDController) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	defer func() {
+		metrics.HandleControllerReconcileMetrics(ExternalDNSCRDControllerName, res, err)
+	}()
+
+	// set up logger
+	logger, err := logr.FromContext(ctx)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("creating logger: %w", err)
+	}
+	logger = ExternalDNSCRDControllerName.AddToLogger(logger).WithValues("namespace", req.Namespace, "name", req.Name)
+
+	obj := &v1alpha1.ExternalDNS{}
+	if err = e.client.Get(ctx, req.NamespacedName, obj); err != nil {
+		logger.Error(client.IgnoreNotFound(err), "failed to get externaldns object, will ignore not found error")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// verify serviceaccount
+	if _, err = keyvault.GetServiceAccountAndVerifyWorkloadIdentity(ctx, e.client, obj.GetInputServiceAccount(), obj.GetNamespace()); err != nil {
+
+		var userErr keyvault.UserError
+		if errors.As(err, &userErr) {
+			logger.Info("failed to verify service account due to user error, sending warning event: " + userErr.UserError())
+			e.events.Eventf(obj, corev1.EventTypeWarning, "FailedUpdateOrCreateExternalDNSResources", "failed serviceaccount verification: %s", userErr.UserError())
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "failed to verify service account")
+		return ctrl.Result{}, err
+	}
+
+	inputDNSConf := buildInputDNSConfig(obj)
+	manifestsConf, err := manifests.NewExternalDNSConfig(e.config, inputDNSConf)
+	if err != nil {
+		logger.Error(err, "failed to generate ExternalDNS resources from ExternalDNS CR")
+		e.events.Eventf(obj, corev1.EventTypeWarning, "FailedUpdateOrCreateExternalDNSResources", "failed to generate external DNS resources: %s", err)
+		return ctrl.Result{}, err
+	}
+
+	resources := manifestsConf.Resources()
+
+	multiError := &multierror.Error{ErrorFormat: func(errs []error) string {
+		errStrings := make([]string, 0, len(errs))
+		for err := range errs {
+			errStrings = append(errStrings, errs[err].Error())
+		}
+		return strings.Join(errStrings, ", ")
+	}}
+
+	for _, resource := range resources {
+		resource.SetOwnerReferences([]metav1.OwnerReference{{
+			APIVersion: obj.APIVersion,
+			Controller: util.ToPtr(true),
+			Kind:       obj.Kind,
+			Name:       obj.Name,
+			UID:        obj.UID,
+		}})
+
+		currentResourceErr := util.Upsert(ctx, e.client, resource)
+		if currentResourceErr != nil {
+			logger.Error(currentResourceErr, "failed to upsert externaldns resources")
+		}
+		multiError = multierror.Append(multiError, currentResourceErr)
+	}
+
+	if multiError.ErrorOrNil() != nil {
+		logger.Error(err, "failed to upsert externaldns resources")
+		e.events.Eventf(obj, corev1.EventTypeWarning, "FailedUpdateOrCreateExternalDNSResources", "failed to deploy external DNS resources: %s", multiError.Error())
+		return ctrl.Result{}, errors.New(multiError.Error())
+	}
+
```

Comment:
there's no logic for updating the Status fields of externaldns. I'm look with adding that in future revisions but just want to call that out so we don't forget about them

Reply:
you mean separate from the setcondition method on the crd? https://github.com/jaiveerk/aks-app-routing-operator/blob/3fbeb7179817c1fccdd04ee2c5853a65886ec28b/api/v1alpha1/externaldns_types.go#L45-L45

Reply:
no. There's no logic to set those fields. This is the right controller to do that in